clients/src/main/java/org/apache/kafka/clients/Metadata.java

```diff
@@ -331,6 +339,24 @@ public synchronized void removeListener(Listener listener) {
         this.listeners.remove(listener);
     }
 
+    /**
+     * "Close" this metadata instance to indicate that metadata updates are no longer possible. This is typically used
+     * when the thread responsible for performing metadata updates is exiting and needs a way to relay this information
+     * to any other thread(s) that could potentially wait on metadata update to come through.
+     */
+    public synchronized void close() {
+        this.isClosed = true;
+        this.notifyAll();
+    }
+
+    /**
+     * Check if this metadata instance has been closed. See {@link #close()} for more information.
+     * @return True if this instance has been closed; false otherwise
+     */
+    public synchronized boolean isClosed() {
```

Comment:
Does this need to be public?

Reply:
I have `KafkaProducer` calling into this now, so needs to be public.